08-CMSIS/platform/STM32P103/inc/uart.h

```diff
@@ -0,0 +1,11 @@
+#ifndef __UART_H_         
+#define __UART_H_
+
+
+void usart_init(void);
+void print_str(const char *str);
+
```

Comment:
You can declare serial_io interface as following:
```
typedef struct {
    int (*init)(void *self);
    void (*puts)(void *self, const char *str);
} serial_io;
```

Then, you can specialize the platform-specific implementations for each STM32 variant as following:
```
static int init(...) { ... }
static void puts(...) { ... }
serial_io stm32_p103_serial = {
    .init = init,
    .puts = puts,
};
```